### PR TITLE
Added -<.> instances

### PR DESCRIPTION
**Pull request by dsturnbull**
```
λ> Local "pewp" </> "a" <.> "txt" -<.> "zip"
Local "pewp/a.zip"
```

```
λ> S3 (S3Uri (BucketName "b") (ObjectKey "pewp")) </> "a" <.> "txt" -<.> "zip"
S3 (S3Uri {bucket = BucketName "b", objectKey = ObjectKey "pewp/a.zip"})
```

### CODE REVIEW

File: src/HaskellWorks/Data/Uri/Location.hs [hunk 1]
```diff
@@ -24,11 +24,13 @@ import qualified System.FilePath  as FP
 import qualified Data.Aeson.Types as J
 
 class IsPath a s | a -> s where
-  (</>) :: a -> s -> a
-  (<.>) :: a -> s -> a
+  (</>)  :: a -> s -> a
+  (<.>)  :: a -> s -> a
+  (-<.>) :: a -> s -> a
```
Comment (newhoggy) [inline src/HaskellWorks/Data/Uri/Location.hs]: This operator deserves docs I think 😁

Comment (newhoggy) [reply src/HaskellWorks/Data/Uri/Location.hs]: Is `-<.>` exactly the same as `<.>`?

Comment (newhoggy) [reply src/HaskellWorks/Data/Uri/Location.hs]: Oh, right.  It makes sense now.  Most instance implementations just delegate so they looks the same.

**Comment by newhoggy** [comment]
👍

File: src/HaskellWorks/Data/Uri/Location.hs (hunk 1 above)
Comment (dsturnbull) [reply src/HaskellWorks/Data/Uri/Location.hs]: I think the docs are implicit from seeing this in another path library and wanting to use it here

**Pull request merged**